adoc-mode.el

```diff
@@ -248,6 +257,56 @@ See for example `tempo-template-adoc-title-1'."
                  (const :tag "tempo-snippets" tempo-snippets))
   :group 'adoc)
 
+(defcustom adoc-fontify-code-blocks-natively nil
+  "When non-nil, fontify code in code blocks using the native major mode.
+This only works for code blocks where the language is
+specified where we can automatically determine the appropriate
+mode to use.  The language to mode mapping may be customized by
+setting the variable `adoc-code-lang-modes'."
+  :group 'adoc
+  :type 'boolean
+  :safe 'booleanp)
+
+;; This is based on `org-src-lang-modes' from org-src.el
+(defcustom adoc-code-lang-modes
+  '(("ocaml" . tuareg-mode)
```

Comment:
Those are still not sorted alphabetically.

Reply:
Maybe, the pair `("ocaml" . tuareg-mode)` is listed at the beginning because it is mentioned in the comment? But, sorting alphabetically is fine with me. I'll do it.

Reply:
Hold on. No, `calc` would go between `C` and `cpp`. That does not make sense. I will not keep the sorting at such places.

Reply:
Ah, so this is just mappings that we can't figure out manually. I missed this part the first time I read the code. 

Btw, I'm a bit confused by your comments:

1. Do we even need to consider things like `calc` and `dot`? I thought we wanted to focus on programming languages? 
2. There's nothing special about `tuareg-mode`, other than it's not built-in (but a lot of major modes are not built-in). Btw, in some cases there are multiple major modes that exist for some language - e.g. there's `caml-mode` and `tuareg-mode`. Anyways, this is something that we can fix later.

Reply:
The only special thing of `("ocaml" . tuareg-mode)` is that it is used in the doc string for explaining the purpose of `adoc-code-lang-modes`.

Reply:
Yeah, but all modes in this list have non-standard names (that can't be derived automatically). That's why I still don't get what's special about `tuareg-mode`. Also I don't see the need for mappings to fundamental-mode, as this can be considered some fallback by default.